openmetadata-ui/src/main/resources/ui/src/pages/TasksPage/shared/DiffViewNew.tsx

```diff
@@ -66,50 +66,73 @@ export const DiffViewNew = ({
     return () => clearTimeout(timer);
   }, [diffArr]);
 
-  const elements = diffArr.map((diff) => {
+  const getDiffKey = (diff: Change, index: number) => {
     if (diff.added) {
-      return (
-        <ins
-          className="diff-added-new"
-          data-testid="diff-added"
-          key={uniqueId()}>
-          <TaskDescriptionPreviewer
-            enableSeeMoreVariant={false}
-            markdown={diff.value}
-            showReadMoreBtn={false}
-          />
-        </ins>
-      );
+      return `diff-${index}-added`;
     }
     if (diff.removed) {
-      return (
-        <del
-          className="diff-removed-new"
-          data-testid="diff-removed-new"
-          key={uniqueId()}>
-          <TaskDescriptionPreviewer
-            enableSeeMoreVariant={false}
-            markdown={diff.value}
-            showReadMoreBtn={false}
-          />
-        </del>
-      );
+      return `diff-${index}-removed`;
     }
 
-    return (
-      <span
-        className="diff-normal-new"
-        data-testid="diff-normal-new"
-        key={uniqueId()}>
-        {' '}
-        <TaskDescriptionPreviewer
-          enableSeeMoreVariant={false}
-          markdown={diff.value}
-          showReadMoreBtn={false}
-        />
-      </span>
-    );
-  });
+    return `diff-${index}-normal`;
+  };
+
+  const elements = useMemo(
+    () =>
+      diffArr.map((diff, index) => {
+        const key = getDiffKey(diff, index);
+
```

Comment:
@shrushti2000 apart from the index don't we have any other thing that we use inplace of index ?

Reply:
Previously when we  were using uniqueId() for the key it  was causing  some re rendering issue. The Diff View was getting rendered on every interaction

Reply:
Understood but don't we have any unique value from `diff` that we can use and avoid index here?

Reply:
we can use value right here, since it will be different for every one ?

Reply:
updated the key with diff-${diff.value}-${diff.removed}-${diff.added}-added` combination